pkg/agents/team3/agentMain.go

```diff
@@ -74,38 +87,41 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
 }
 
 func (a *CustomAgent3) Run() {
-	//Update agent variables at the beginning of day (when HP has been reduced)
-	if a.HP() < a.knowledge.lastHP {
-		changeNewDay(a)
+	//Update agent variables at the beginning of day (when we age)
+	if a.knowledge.agentAge < a.BaseAgent().Age() {
+		a.changeNewDay()
 	}
 
 	//Update agent variables at the beginning of reshuffle (when floor has changed)
 	if len(a.knowledge.floors) == 0 || a.knowledge.floors[len(a.knowledge.floors)-1] != a.Floor() {
-		changeNewFloor(a)
+		a.changeNewFloor()
 	}
-	a.Log("Custom agent 3 each run:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "mood": a.vars.mood, "morality": a.vars.morality})
+	a.Log("Agent 3 each run:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "mood": a.vars.mood, "morality": a.vars.morality, "stubbornness": a.vars.stubbornness})
 
-	//eat
-	foodTaken, err := a.TakeFood(food.FoodType((a.takeFoodCalculation())))
-	if err != nil {
-		switch err.(type) {
-		case *infra.FloorError:
-		case *infra.NegFoodError:
-		case *infra.AlreadyEatenError:
-		default:
+	//Try to take food every time RIP
+	if a.CurrPlatFood() != -1 {
+		foodTaken, err := a.TakeFood(food.FoodType((a.takeFoodCalculation())))
+		if err != nil {
+			switch err.(type) {
+			case *infra.FloorError:
+			case *infra.NegFoodError:
+			case *infra.AlreadyEatenError:
+			default:
+			}
+		} else {
+			//Update variables right after eating
+			a.knowledge.foodLastSeen = a.BaseAgent().CurrPlatFood()
+			a.knowledge.lastHP = a.HP()
+			a.knowledge.foodLastEaten = food.FoodType(foodTaken)
+			a.decisions.foodToEat = -1
+			a.decisions.foodToLeave = -1
 		}
-	} else {
-		//Update variables right after eating
-		a.knowledge.lastHP = a.HP()
-		a.knowledge.foodLastEaten = food.FoodType(foodTaken)
-		a.decisions.foodToEat = -1
-		a.decisions.foodToLeave = -1
 	}
 
-	a.message()
-
-	//eat
-
-	//send Message
+	if a.knowledge.agentAge == 0 { //happens only on the first day
+		a.knowledge.foodMovingAvg = float64(a.foodReqCalc(50, 50))
```

Comment:
You can put this in teh constructor

Reply:
Since foodReqCalc is a member function, we were not able to do this

Reply:
It requires info from the agent